opal/corelib/module.rb

```diff
@@ -645,8 +683,18 @@ def to_s
 
   def undef_method(*names)
     %x{
-      for (var i = 0, length = names.length; i < length; i++) {
-        Opal.udef(self, "$" + names[i]);
+      var length = names.length;
+
+      if (length > 0) {
```

Comment:
Seems like an extraneous check here too... why isn't undef_method an alias to remove_method? Does it differ somehow?

Reply:
One calls Opal._r_def, one calls Opal._u_def

Reply:
Check removed